src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -418,7 +418,8 @@ impl CatalogState {
 
         if !entry.item().is_temporary() {
             // Populate or clean up the `mz_object_dependencies` table.
-            for dependee in &entry.item().uses().0 {
+            // TODO(jkosh44) Unclear if this table wants to include all uses or only references.
```

Comment:
This table is used to populate pg_depends which cares about being able to DROP stuff and CASCADE. Presumably that would mean this should be `uses` so it includes all of the not explicit objects as well, because those should also cause a DROP to fail.

I also wonder if...`.references()` just shouldn't exist and it should always be `.uses()`? Are there known instances where we definitely want dependency group 2 but not 3?

Reply:
> I also wonder if....references() just shouldn't exist and it should always be .uses()? Are there known instances where we definitely want dependency group 2 but not 3?

The one I can remember off the top of my head is for RBAC. There's an argument that you should only need privileges on objects that are directly referenced and not on objects that may be used to implement some parts of your query. Though there's maybe an argument for the other side as well.

Reply:
Another one I just remembered is for renaming. If object `A` is renamed then we only need to update objects that reference `A`, not all objects that use `A`. We could just update all objects that use `A`, and it will be a no-op for the objects that don't actually reference `A` though.

Reply:
I also see that the sink comments use `.references()`, but I can't remember why.

Reply:
Ok. I think things are good and we should try to get this in now. It seems that only these above things changed, and mostly stuff is still calling `uses()`. Thanks.